### DISCUSSION TITLE
How are gene activity scores computed?

### DISCUSSION THREAD

**Discussion by jules-samaran** [stuart-lab/signac#1132] [Q&A] [answered]
Hi,

Thanks for your work on this, Signac is being very relevant to my research.
I am currently using it to compute gene activity scores on scATAC-seq data and I had a few questions whose answers I couldn't find in the docs:
- does the way you count fragments that fall in the promoter region account for the transcription direction (which is symmetric between the two strands)? I see the user can specify the "number of bases to extend upstream (and downstream) of the TSS" and I was wondering if this notion of upstream related to the transcription direction or if it was actually the same direction for the two strands (which could be a problem).
- what could explain the fact that I don't obtain gene activity scores for some coding gene that are measured in scRNA-seq experiments ? Does it mean that no fragment at all came from the gene's body and promoter region and that then it should be a zero?
For example in one of my experiments I had gene activity scores for approximately 19k genes but when I compared them with data from an scRNA-seq experiment (on the same cells), more than 1000 genes were missing from the gene activity table, some of them being coding genes such as HDAC9.

**Comment by timoast** [comment, marked as answer]
> does the way you count fragments that fall in the promoter region account for the transcription direction (which is symmetric between the two strands)? I see the user can specify the "number of bases to extend upstream (and downstream) of the TSS" and I was wondering if this notion of upstream related to the transcription direction or if it was actually the same direction for the two strands (which could be a problem).

Yes, upstream/downstream is dependent on the strand

> what could explain the fact that I don't obtain gene activity scores for some coding gene that are measured in scRNA-seq experiments ? Does it mean that no fragment at all came from the gene's body and promoter region and that then it should be a zero?

If the gene name is present in the output matrix but there are no counts for that gene, it's likely there were no fragments aligned to the gene body and upstream 2 kb region for that gene. Alternatively, if the gene name is missing from the matrix, it's likely not present in the gene annotations you're using or it's not classified as protein coding.

If there are particular genes that you're interested in, I'd suggest looking at those using `CoveragePlot()` to verify that there are fragments aligned to the region that's being quantified by `GeneActivity()`. You can adjust the `extend.upstream`,`extend.downstream`,`biotypes`, and `max.width` parameters to capture genes you might be missing.
